src/js/medium-editor.js

```diff
@@ -1220,6 +1220,11 @@ else if (typeof define === 'function' && define.amd) {
 
                     self.createLink(this, target, button);
                 }
+                else if (e.keyCode === 27) {
+                    e.preventDefault();
+                    self.showToolbarActions();
+                    restoreSelection.call(self, self.savedSelection);
```

Comment:
If you're suggesting to DRY it out by creating a single function that can be called by all UI actions intended to do the same thing, then I'm all for it. 

Something like this? In this case, it's a two-line function, which may or may not be worthy of factoring out. 

    function restoreEditorState() {
        this.showToolbarActions();
        this.restoreSelection(this.savedSelection);
    }